src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs

```diff
@@ -236,6 +301,80 @@ protected static async Task<ImmutableArray<SyntaxToken>> GetIdentifierOrGlobalNa
             };
         }
 
+        protected static async Task<ImmutableArray<FinderLocation>> FindReferencesInDocumentInsideGlobalSuppressionsAsync(
+            Document document,
+            SemanticModel semanticModel,
+            ISyntaxFacts syntaxFacts,
+            Func<string> getDocumentationCommentId,
+            CancellationToken cancellationToken)
+        {
+            var info = await SyntaxTreeIndex.GetIndexAsync(document, cancellationToken).ConfigureAwait(false);
+            if (!info.ContainsGlobalAttributeInfo)
+            {
+                return ImmutableArray<FinderLocation>.Empty;
+            }
+
+            var suppressMessageAttribute = semanticModel.Compilation.SuppressMessageAttributeType();
```

Comment:
do this check first?

Reply:
I expect the attribute to be non-null for almost all realistic compilations, but the global attribute info cache will be empty for majority of source files.